src/Databases/DatabaseFactory.cpp

```diff
@@ -96,6 +97,9 @@ DatabasePtr DatabaseFactory::getImpl(const ASTCreateQuery & create, const String
         throw Exception("Database engine " + engine_name + " cannot have parameters, primary_key, order_by, sample_by, settings",
                         ErrorCodes::UNKNOWN_ELEMENT_IN_AST);
 
+    if (context.hasQueryContext() && context.getSettingsRef().log_queries)
```

Comment:
BTW, will it record non-existent database engines (before throwing exception)?
It should not behave this way. Let's also check other factories.